cmd/juju/application/bundle/bundle_test.go

```diff
@@ -233,11 +234,35 @@ func (s *composeAndVerifyRepSuite) TestComposeAndVerifyBundleOverlay(c *gc.C) {
 		"blog-title": "magic bundle config",
 	}
 
-	obtained, err := ComposeAndVerifyBundle(s.bundleDataSource, []string{s.overlayFile})
+	obtained, _, err := ComposeAndVerifyBundle(s.bundleDataSource, []string{s.overlayFile})
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(obtained, gc.DeepEquals, &expected)
 }
 
+func (s *composeAndVerifyRepSuite) TestComposeAndVerifyBundleOverlayUnmarshallErrors(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+	bundleData, err := charm.ReadBundleData(strings.NewReader(typoBundle))
+	c.Assert(err, jc.ErrorIsNil)
+	expectedError := errors.New("unmarshal document 0: yaml: unmarshal errors:\n  line 1: field sries not found in bundle\n  line 18: field constrai not found in machines")
```

Comment:
Is there a better error message we could be giving for these?
To me this reads like Juju wants a field named "sries" but it couldn't find it. Consider an error like:
```
  line 1: unrecognized field "sries"
```
I'm guessing we're getting the error from the underlying library, so this may not be easy. But these feel really hard to understand as an end user.